### PR TITLE
fix: simplify stream re-auth

### PR DESCRIPTION
**Pull request by thisbejim**
Issue: https://github.com/thisbejim/Pyrebase/issues/80

### CODE REVIEW

File: pyrebase/pyrebase.py [hunk 1]
```diff
@@ -553,9 +555,8 @@ def start(self):
     def start_stream(self):
         self.sse = ClosableSSEClient(self.url, session=self.make_session(), build_headers=self.build_headers)
         for msg in self.sse:
-            msg_data = json.loads(msg.data)
-            # don't return initial data
-            if msg_data:
+            if msg:
+                msg_data = json.loads(msg.data)
```
Comment (aleksanb) [inline pyrebase/pyrebase.py]: You will need to check whether msg_data is something as well, as "msg.data" can be "null", which is a message, but evaluates to python "None", which will crash on line 560 when you try to assign to it.
Here's what we've done: https://github.com/comoyo/Pyrebase/commit/81ed523407ab0f3ceb7f6168bcc1af23ca36938e.

**Comment by thisbejim** [comment]
Thanks for the code review @aleksanb, I really appreciate it! I actually discovered this issue too and this is actually being handled inside sseclient in this PR. If msg.data is 'null' we return None to the stream_handler instead of the msg.

**Comment by busla** [comment]
It can also contain the string _credentials are invalid_ so maybe checking if its a dict might be a better handler.

if isinstance(msg.data, dict):
    ....

**Comment by thisbejim** [comment]
Thanks @bulsa, I'm checking for that too in sseclient in the PR.

File: sseclient/sseclient.py [hunk 2]
```diff
@@ -97,6 +81,13 @@ def __next__(self):
         self.buf = tail
         msg = Event.parse(head)
 
+        if msg.data == "credential is no longer valid":
```
Comment (thisbejim) [inline sseclient/sseclient.py]: Here are the sseclient changes.

**Comment by busla** [comment]
sorry mate, didn´t look at the PR 😄

If we check for dict type, we wont have to explicitly check for the value, wether it´s a string, null or whatever.

We would always want a non-empty dict, since a ref is being updated, right?

**Pull request merged**